### PR TITLE
Add the "cancelled" reason for rejection in docs

### LINKED ISSUE #36894: "API Priority and Fairness" doc must be updated
(1) The user-facing changes in https://github.com/kubernetes/kubernetes/pull/110104 are not yet reflected in the doc.

(2) There are four possible values for the `reason` of a rejection ("concurrency-limit", "queue-full", "time-out", and "cancelled") but "cancelled" is not listed in the doc.

The relevant page is https://kubernetes.io/docs/concepts/cluster-administration/flow-control/

/sig api-machinery
@tkashem 
@MikeSpreitzer

### PR DESCRIPTION
**Pull request by Rishit-dagli**
In the "API Priority and Fairness" documentation, the "cancelled" reason for rejection is not documented. This PR adds this to the documentation.

Closes #36894

### CODE REVIEW

File: content/en/docs/concepts/cluster-administration/flow-control.md [hunk 1]
```diff
@@ -450,6 +450,8 @@ poorly-behaved workloads that may be harming system health.
     queue excess requests, or
   * `time-out`, indicating that the request was still in the queue
     when its queuing time limit expired.
+  * `cancelled`, indicating that the request it is not purge locked
+    and has been ejected from the queue.
```
Comment (tengqm) [inline content/en/docs/concepts/cluster-administration/flow-control.md:454]: Please revise the sentence. It doesn't flow well...

File: content/en/docs/concepts/cluster-administration/flow-control.md [hunk 2]
```diff
@@ -450,6 +450,8 @@ poorly-behaved workloads that may be harming system health.
     queue excess requests, or
   * `time-out`, indicating that the request was still in the queue
     when its queuing time limit expired.
+  * `cancelled`, indicating that the request it is not purge locked
```
Comment (divya-mohan0209) [inline content/en/docs/concepts/cluster-administration/flow-control.md:453]:
```suggestion
  * `cancelled`, indicating that the request is not purge locked
```

**Review by divya-mohan0209 — CHANGES_REQUESTED** [review]
@Rishit-dagli : PTAL & advise if you're able to work on the changes?

**Comment by kbhawkey** [comment]
👋 @Rishit-dagli .
Can you accept the suggested changes or do you have questions?

File: content/en/docs/concepts/cluster-administration/flow-control.md (hunk 2 above)
Comment (Rishit-dagli) [reply content/en/docs/concepts/cluster-administration/flow-control.md:453]: Ah, a typo

File: content/en/docs/concepts/cluster-administration/flow-control.md (hunk 1 above)
Comment (Rishit-dagli) [reply content/en/docs/concepts/cluster-administration/flow-control.md]: Fixed with the latest commit

**Comment by Rishit-dagli** [comment]
> 👋 @Rishit-dagli . Can you accept the suggested changes or do you have questions?

Just slipped out of my mind, thanks for the reminder

**Comment by divya-mohan0209** [comment]
@kubernetes/sig-api-machinery-api-reviews : PTAL

**Comment by sftim** [comment]
LGTM for docs

Normally I'd want a tech review from the API machinery SIG; I'll take https://github.com/kubernetes/website/issues/36894 as confirmation that this change is right.

/lgtm

**Comment by tengqm** [comment]
> LGTM for docs
> 
> Normally I'd want a tech review from the API machinery SIG; I'll take #36894 as confirmation that this change is right.
> 
> /lgtm

I did a quick grep on the source code and confirmed that this change is good.

**Pull request merged**